decorators_library/decorators.py

```diff
@@ -1 +1,132 @@
-# implement your decorators here.
+import time
+import inspect
+from .exceptions import TimeoutError
+from logging import getLogger
+
+class Timeout(object):
+    def __init__(self, func, max_time):
+        self.func = func
+        self.max_time = max_time
+    
+    def __call__(self, *args, **kwargs):
+        start = time.time()
+        return_val = self.func(*args, **kwargs)
+        end = time.time()
+        
+        if end - start > self.max_time:
+            raise TimeoutError('Function call timed out')
+        return return_val
+
+def timeout(max_time):
+    def decorator(func):
+        return Timeout(func, max_time)
+    return decorator
+    
+def debug(logger=None):
+    def other(func):
+        def wrapper(*args, **kwargs):
+            return_val = func(*args, **kwargs)
+            if logger is not None:
+                log = logger
+            else:  
+                log = getLogger('tests.test_decorators')
+                
```

Comment:
def decorate(func):
      log = logger if logger else logging.getLogger('tests.test_decorators')

as a separate func would've also worked and be shorter / potentially more pythonic i think